java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,750 @@
+    currentBlock = createBranch(cond, thenBlock, elseBlock);
+    buildCondition(cond.condition(), thenBlock, elseBlock);
+  }
+
+  private void buildVariable(VariableTree tree) {
+    currentBlock.elements.add(tree);
+    ExpressionTree initializer = tree.initializer();
+    if (initializer != null) {
+      build(initializer);
+    }
+  }
+
+  private void buildBinaryExpression(Tree tree) {
+    BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) tree;
+    currentBlock.elements.add(tree);
+    build(binaryExpressionTree.rightOperand());
+    build(binaryExpressionTree.leftOperand());
+  }
+
+  private void buildAssignment(AssignmentExpressionTree tree) {
+    currentBlock.elements.add(tree);
+    build(tree.variable());
+    build(tree.expression());
+  }
+
+  private void buildMemberSelect(MemberSelectExpressionTree tree) {
+    MemberSelectExpressionTree mse = tree;
+    currentBlock.elements.add(mse);
+    // int.class or String[].class are memberSelectExpression which expression part is not an expression.
+    if (!"class".equals(mse.identifier().name())) {
+      build(mse.expression());
+    }
+  }
+
+  private void buildConditionalAnd(BinaryExpressionTree tree) {
+    // process RHS
+    Block falseBlock = currentBlock;
+    currentBlock = createBlock(falseBlock);
+    build(tree.rightOperand());
+    Block trueBlock = currentBlock;
+    // process LHS
+    currentBlock = createBranch(tree, trueBlock, falseBlock);
+    build(tree.leftOperand());
+  }
+
+  private void buildConditionalOr(BinaryExpressionTree tree) {
+    // process RHS
+    Block trueBlock = currentBlock;
+    currentBlock = createBlock(trueBlock);
+    build(tree.rightOperand());
+    Block falseBlock = currentBlock;
+    // process LHS
+    currentBlock = createBranch(tree, trueBlock, falseBlock);
+    build(tree.leftOperand());
+  }
+
+  private void buildLabeledStatement(LabeledStatementTree tree) {
+    LabeledStatementTree s = tree;
+    build(s.statement());
+    currentBlock = createBlock(currentBlock);
+    labels.put(s.label().name(), currentBlock);
+    return;
+  }
+
+  private void buildSwitchStatement(SwitchStatementTree tree) {
+    // FIXME useless node created for default cases.
+    SwitchStatementTree switchStatementTree = tree;
+    Block switchSuccessor = currentBlock;
+    // process condition
+    currentBlock = createBlock();
+    currentBlock.terminator = switchStatementTree;
+    switches.addLast(currentBlock);
+    build(switchStatementTree.expression());
+    // process body
+    currentBlock = createBlock(switchSuccessor);
+    breakTargets.addLast(switchSuccessor);
+    if (!switchStatementTree.cases().isEmpty()) {
+      CaseGroupTree firstCase = switchStatementTree.cases().get(0);
+      for (CaseGroupTree caseGroupTree : Lists.reverse(switchStatementTree.cases())) {
+        build(caseGroupTree.body());
+        switches.getLast().successors.add(currentBlock);
+        if (caseGroupTree != firstCase) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)